app/src/test/java/com/arygm/quickfix/model/categories/PlumbingSubCategoryTest.kt

```diff
@@ -0,0 +1,49 @@
+package com.arygm.quickfix.model.categories
+
+import com.arygm.quickfix.model.categories.plumbing.PlumbingCategory
+import com.arygm.quickfix.model.categories.plumbing.PlumbingSubCategory
+import org.junit.Assert.*
+import org.junit.Test
+
+class PlumbingSubCategoryTest {
+
+  @Test
+  fun testAllEnumValues() {
+    val values = PlumbingSubCategory.entries.toTypedArray()
+    assertEquals(14, values.size)
+
+    for (subCategory in values) {
+      assertNotNull(subCategory)
+      assertNotNull(subCategory.parentCategory)
+      assertNotNull(subCategory.displayName)
+    }
+  }
+
+  @Test
+  fun testValueOf() {
+    val rootRemoval = PlumbingSubCategory.valueOf("ROOT_REMOVAL")
+    assertEquals(PlumbingSubCategory.ROOT_REMOVAL, rootRemoval)
+
+    val installation = PlumbingSubCategory.valueOf("INSTALLATION")
+    assertEquals(PlumbingSubCategory.INSTALLATION, installation)
+  }
+
+  @Test
+  fun testEnumProperties() {
+    val officeBuildingsPlumbing = PlumbingSubCategory.OFFICE_BUILDINGS_PLUMBING
+    assertEquals(PlumbingCategory.COMMERCIAL_PLUMBING, officeBuildingsPlumbing.parentCategory)
+    assertEquals("Office Buildings Plumbing", officeBuildingsPlumbing.displayName)
+  }
+
+  @Test
+  fun testGroupingByParentCategory() {
+    val pipeInspectionSubCategories =
+        PlumbingSubCategory.values().filter {
+          it.parentCategory == PlumbingCategory.PIPE_INSPECTION
+        }
+    val expectedSubCategories =
+        listOf(PlumbingSubCategory.VIDEO_INSPECTION, PlumbingSubCategory.ROOT_REMOVAL)
+    assertEquals(expectedSubCategories.size, pipeInspectionSubCategories.size)
+    assertTrue(pipeInspectionSubCategories.containsAll(expectedSubCategories))
+  }
+}
```

Comment:
For Category and SubCategory tests, wouldn't it be better to have tests for a general category (maybe create an abstract class for tests) rather than a specific one (same for subcategories)? Otherwise, we'll end up with far too many tests.

Reply:
I specified in commit message that I will change these tests and modify them, don't worry about it I will focus on them later on. By that I mean that I will work on a hole pr that will add all the categories and their subcategories as well and test them accordingly.